src/ast/nodes/Import.ts

```diff
@@ -81,11 +81,11 @@ export default class Import extends NodeBase {
 		if (importMechanism) {
 			const leftMechanism =
 				(this.resolutionInterop && importMechanism.interopLeft) || importMechanism.left;
-			code.overwrite(this.parent.start, this.parent.arguments[0].start, leftMechanism);
+			code.overwrite(this.parent.start, this.parent.callee.end, leftMechanism);
```

Comment:
In the initial version, this would also overwrite the opening `(` which enabled nicer output for AMD. There is actually a helper to detect substrings outside comments that could help you fix this: `findFirstOccurrenceOutsideComment` in `utils/renderHelpers.ts`. Here is how you could use it:

```js
// ...
const leftMechanism = (this.resolutionInterop && importMechanism.interopLeft) || importMechanism.left;
const leftMechanismEnd = findFirstOccurrenceOutsideComment(code.original, '(', this.parent.callee.end) + 1;
code.overwrite(this.parent.start, leftMechanismEnd, leftMechanism);
// ...
```

Basically this function receives a string in which to search (which is usually `code.original`), a start index from which to search (`this.parent.callee.end` is the latest I can think of), and a string to search for, and it will return the position in the string.

With this change, no adaptions to the mechanisms should be necessary any more (unless I overlook something).

Reply:
Thanks a lot for the feedback, i will update the PR! I was not very happy either with the: `require[('./generated.')]`

But i was not able to find a better solution that didn't generate invalid JS